miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/WorkflowRestController.java

```diff
@@ -67,4 +79,28 @@
   public List<WorkflowStateDto> list() throws IOException {
     return workflowManager.listUserWorkflows().stream().map(Dtos::asDto).collect(Collectors.toList());
   }
+
+  @RequestMapping(method = RequestMethod.POST, value = "/favourites/add/{workflowName}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @ResponseBody
+  public void addFavourite(@PathVariable WorkflowName workflowName)
+      throws IOException {
+    User user = securityManager.getUserById((long) 1);
+    Set<WorkflowName> favouriteWorkflows = user.getFavouriteWorkflows();
+    favouriteWorkflows.add(workflowName);
+    user.setFavouriteWorkflows(favouriteWorkflows);
```

Comment:
This set (and the one on line 103 are unnecessary. The same collection is still in the object.